From 68587d886984bbae2e7ea6f419d5feabcd24a53e Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime@cerno.tech>
Date: Fri, 6 May 2022 12:24:31 +0200
Subject: [PATCH] clk: Add locking to clk_get_rate_range

clk_get_rate_range() is missing the proper locks, making lockdep
complain.

Signed-off-by: Maxime Ripard <maxime@cerno.tech>
---
 drivers/clk/clk.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -2588,7 +2588,9 @@ void clk_get_rate_range(struct clk *clk,
 	if (!clk || !min || !max)
 		return;
 
+	clk_prepare_lock();
 	clk_core_get_boundaries(clk->core, min, max);
+	clk_prepare_unlock();
 }
 EXPORT_SYMBOL_GPL(clk_get_rate_range);
 
